pkg/server/api/middleware/ratelimit.go

```diff
@@ -255,26 +258,41 @@ func logLimiterMisuse(ctx context.Context, rateLimiter api.RateLimiter, used boo
 type rateLimiterWrapper struct {
 	rateLimiter api.RateLimiter
 	used        bool
+	metrics     telemetry.Metrics
 }
 
-func (w *rateLimiterWrapper) RateLimit(ctx context.Context, count int) error {
+func (w *rateLimiterWrapper) RateLimit(ctx context.Context, count int) (err error) {
 	w.used = true
+	switch w.rateLimiter.(type) {
+	case *perCallLimiter, *perIPLimiter:
+		counter := telemetry.StartCall(w.metrics, "rateLimit", getNames(ctx)...)
+		defer counter.Done(&err)
+	}
```

Comment:
I think this might be more robust if we emit metrics by default. Maybe we define a new interface:
```
type noopRateLimiter struct {
    noop()
}
```

Then both `noLimit` and `disabledLimit` gain a `noop()` method.

Then we can do a simple test here to see if we should be emitting metrics:
```suggestion
	if _, noop := w.rateLimiter.(noopRateLimiter); !noop {
		counter := telemetry.StartCall(w.metrics, "rateLimit", getNames(ctx)...)
		defer counter.Done(&err)
	}
```